### DISCUSSION TITLE
Imported data can edit only same admin whose has import that data. Not every admin can edit that data.

### DISCUSSION THREAD

**Discussion by Pratikshakhandagale** [Sunbird-RC/community#277] [Issues]
Right now, Imported data can edit only same admin whose has import that data. Not every admin can edit or view it. 

Expected result is - Each admin able be update and view uploaded data. 

cc - @tejash-jl

**Comment by Pratikshakhandagale** [comment]
@tejash-jl @dileepbapat Can you please help us in above scenario, we need it urgently.

**Reply by tejash-jl** [reply]
Yes, I'm working on it

**Reply by Pratikshakhandagale** [reply]
@tejash-jl   Any Updates?

**Comment by parthlawate** [comment]
So this is a access control /Policy related ? 

I assume when you say Import, we are calling the create API. What we are saying is that when create api is called with certain credentials, the records are owned by that user... How do we setup users & access control that allows certain users to edit those records & also in the future add them on behalf of someone else  (Assisted Registry onboarding/ Edit scenarios) 

Another question - As per my understanding, we have Activity logs/ trails in Sunbird RC. What APIs are needed to show these ? Specifically in the case when a user other than data owner is editing, these trails would be important. 

cc @baljeetsingh

**Reply by tejash-jl** [reply]
We used to capture the ownership of the entity based on the `ownershipAttribute` or `roles` config specified in the schema config. 
In prathiksha scenario they have configured `roles`, so we check if the user who is creating the entity has that role in the token passed. And we store that specific user as the owner of that entity. So when a new user with the same role tries to get that data, it'll be forbidden.
Im now modifying to retrieve the data based on roles check and not on ownership if `roles` are configured for that schema.

**Reply by parthlawate** [reply]
got it .. cool !

**Comment by Pratikshakhandagale** [comment]
@tejash-jl  It's working as expected. Thank you !!
